winpr/libwinpr/file/file.c:133:35: warning: left shift count >= width of type [-Wshift-count-overflow]

I get this warning, when I compile the current master for different openSUSE versions; 32bit and 64bit; Tumbleweed and Leap 42.*.
Built with:
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_LIBDIR:PATH=lib -DCMAKE_SKIP_INSTALL_RPATH=ON -DWITH_CUPS=ON -DWITH_CLIENT=ON -DWITH_PCSC=ON -DWITH_ALSA=ON -DWITH_PULSE=ON -DWITH_OSS=OFF -DWITH_DIRECTFB=OFF -DWITH_JPEG=ON -DWITH_GSM=ON -DWITH_GSTREAMER_1_0=ON -DWITH_IPP=OFF -DWITH_OPENSSL=ON -DWITH_CHANNELS=ON -DWITH_X11=ON -DWITH_XCURSOR=ON -DWITH_XEXT=ON -DWITH_XKBFILE=ON -DWITH_XI=ON -DWITH_XINERAMA=ON -DWITH_XRENDER=ON -DWITH_XV=ON -DWITH_ZLIB=ON -DWITH_KRB5=ON -DBUILTIN_CHANNELS=OFF -DWITH_SERVER=OFF -DWITH_SSE2=ON -DWITH_DEBUG_SCARD=OFF -DWITH_SMARTCARD_INSPECT=ON -DWITH_WAYLAND=ON -DWITH_LIBSYSTEMD=ON -DCMAKE_NO_BUILTIN_CHRPATH=ON .

complete warning:
[   19s] /home/abuild/rpmbuild/BUILD/FreeRDP-master/winpr/libwinpr/file/file.c: In function 'FileSetFilePointer':
[   19s] /home/abuild/rpmbuild/BUILD/FreeRDP-master/winpr/libwinpr/file/file.c:133:35: warning: left shift count >= width of type [-Wshift-count-overflow]
[   19s]    offset = (*lpDistanceToMoveHigh << 32) | (UINT32)lDistanceToMove;
[   19s]                                    ^~

I guess, this is caused by #4076.

--- a/winpr/libwinpr/file/file.c
+++ b/winpr/libwinpr/file/file.c
@@ -128,9 +128,11 @@ static DWORD FileSetFilePointer(HANDLE hFile, LONG lDistanceToMove,
 	if (!hFile)
 		return INVALID_SET_FILE_POINTER;
 
+	/* If there is a high part, the sign is contained in that
+	 * and the low integer must be interpreted as unsigned. */
 	if (lpDistanceToMoveHigh)
 	{
-		offset = (*lpDistanceToMoveHigh << 32) | (UINT32)lDistanceToMove;
+		offset = (INT64)(((UINT64)*lpDistanceToMoveHigh << 32U) | (UINT64)lDistanceToMove);
 	}
 	else
 		 offset = lDistanceToMove;

